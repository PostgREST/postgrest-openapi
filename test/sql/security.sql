-- shows the JWT security scheme
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'securitySchemes'->'JWT');

-- lists the JWT scheme as a security requirement object 
select get_postgrest_openapi_spec('{public}')->'security' <@ '[{"JWT": []}]'::jsonb as has_jwt;
