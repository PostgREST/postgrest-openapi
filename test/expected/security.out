-- shows the JWT security scheme
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'securitySchemes'->'JWT');
                                        jsonb_pretty                                        
--------------------------------------------------------------------------------------------
 {                                                                                         +
     "type": "http",                                                                       +
     "scheme": "bearer",                                                                   +
     "description": "Adds the JSON Web Token to the `Authorization: Bearer <JWT>` header.",+
     "bearerFormat": "JWT"                                                                 +
 }
(1 row)

-- lists the JWT scheme as a security requirement object 
select get_postgrest_openapi_spec('{public}')->'security' <@ '[{"JWT": []}]'::jsonb as has_jwt;
 has_jwt 
---------
 t
(1 row)

